### PR TITLE
Update docs for contributing 

### PR DESCRIPTION
**Pull request by rramkumar1**
This PR organizes our contributor docs a bit better and removes lots of stale info. Specifically, there is now more information on:

1. how to get your dev environment setup
2. how to run tests and build 
3. how to run the controller locally to test changes. 

There are also other misc. updates. 

Ref: #557

### CODE REVIEW

**Comment by rramkumar1** [comment]
@spencerhance Please take a look and leave comments. Would be great if you could walk through some of the steps to make sure they make sense + work.

File: docs/contrib/dev-setup.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+All ingress-gce binaries are built through a Makefile. Depending on your
+requirements you can build a raw binary, a local container image,
+or push an image to a remote repository. To build all binaries, run:
+
+```console
+make build
+```
+
+The resulting binaries can be found in bin/amd64/. To push an image up to a
+repository, run the following:
```
[the remark this answers is not in the document]
Comment (rramkumar1) [reply docs/contrib/dev-setup.md]: Done

File: docs/deploy/gce/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Then, clone the kubernetes repo and test-infra repo:
+
+```console
+$ cd $GOPATH/src/
+$ git clone https://github.com/kubernetes/kubernetes.git
+$ git clone https://github.com/kubernetes/test-infra.git
+```
+
+Then, build the release tars:
```
[the remark this answers is not in the document]
Comment (rramkumar1) [reply docs/deploy/gce/README.md]: Done

File: docs/contrib/cluster-setup.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+This doc outlines the steps needed to setup a dev cluster for testing changes
+made to the Ingress-GCE controller. Typically we test out controller changes
+by running the binary locally but this still means we need a working cluster
+to test against.
+
+## Create the cluster
+
+We recommend to create a k8s cluster in GCE. For instructions on how to do
+that, go [here][../deploy/gce/README.md]
```
[the remark this answers is not in the document]
Comment (rramkumar1) [reply docs/contrib/cluster-setup.md]: Done

File: docs/deploy/gke/README.md [hunk 4]
```diff
@@ -29,14 +23,14 @@ under [core]/project is the one that contains your cluster. Second, ensure that
 the current logged-in account listed under [core]/account is the owner of the project.
 In other words, this account should have full project ownership permissions and be listed as
 having the "Owner" role on the IAM page of your GCP project. You might ask why this
-is needed? Well, the reason is that the script invokes a kubectl command which
+is needed? Well, the reason is that our [script](../resources/gke-self-managed.sh) invokes a kubectl command which
```
Comment (MrHohn) [inline docs/deploy/gke/README.md]: Need to fix this link `../resources/gke-self-managed.sh`

File: docs/contrib/cluster-setup.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+We recommend to create a k8s cluster in GCE. For instructions on how to do
+that, go [here](../deploy/gce/README.md)
+
+## Remove the default controller
+
+Once the cluster is created, we need to delete the existing Ingress-GCE controller
+that comes by default:
+
+```console
+kubectl delete pod l7-lb-controller
```
Comment (MrHohn) [inline docs/contrib/cluster-setup.md]: Humm.. IIUC kubelet on the master will bring this pod back as long as the manifest still exists. We probably need something like
```sh
$ gcloud compute ssh e2e-test-zihongz-master
$ sudo rm /etc/kubernetes/manifests/glbc.manifest
```

File: docs/deploy/gce/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+Finally, create the cluster:
+
+```console
+go run hack/e2e.go --up
+```
+
+Note that the above command automatically destroys any existing cluster that
+was previously created with --up.
```
Comment (MrHohn) [inline docs/deploy/gce/README.md]: Probably link to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md for further details?

File: docs/contrib/cluster-setup.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+```
+
+This ensures we can run our own copy of the controller locally.
```
Comment (MrHohn) [inline docs/contrib/cluster-setup.md]: Would be great to link to `docs/deploy/local/README.md`?

File: docs/deploy/local/README.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+A skeleton configuration file can be found [here](../resources/gce.conf). An
+example fully-specified config looks something like this:
+
+```console
+[global]
+token-url = nil
+project-id = foo-project
+network-name = foo-network
+subnetwork-name = foo-subnetwork
+node-instance-prefix = gke-foo-cluster
```
Comment (MrHohn) [inline docs/deploy/local/README.md]: Humm, what does it looks like in the e2e cluster case? Something like `node-instance-prefix = e2e-test-zihongz`?

**Comment by rramkumar1** [comment]
@MrHohn This got merged before I resolved your comments. Will open another PR with those fixes plus others. Thanks!

**Pull request merged**